test/unit/validate_test.py

```diff
@@ -0,0 +1,175 @@
+    ],
+)
+def test_weighted_quantile_parametrized(data, weights, quantile, expected):
+    """Test the weighted quantile function with various inputs."""
+    data_series = pd.Series(data)
+    weights_series = pd.Series(weights)
+    result = pv.weighted_quantile(data_series, weights_series, quantile)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "data,weights,quantile,expected_error",
+    [
+        # Invalid quantile values
+        ([1, 2, 3], [1, 1, 1], -0.1, ValueError),
+        ([1, 2, 3], [1, 1, 1], 1.1, ValueError),
+        # Mismatched lengths
+        ([1, 2, 3], [1, 1], 0.5, ValueError),
+        ([1, 2], [1, 1, 1], 0.5, ValueError),
+    ],
+)
+def test_weighted_quantile_errors(data, weights, quantile, expected_error):
+    """Test that weighted_quantile raises appropriate errors for invalid inputs."""
+    data_series = pd.Series(data)
+    weights_series = pd.Series(weights)
+    with pytest.raises(expected_error):
+        pv.weighted_quantile(data_series, weights_series, quantile)
+
+
+@pytest.mark.parametrize(
+    "data,weights,quantile",
+    [
+        # All NaN data
+        ([np.nan, np.nan, np.nan], [1, 1, 1], 0.5),
+        # All zero weights
+        ([1, 2, 3], [0, 0, 0], 0.5),
+        # Empty series
+        ([], [], 0.5),
+    ],
+)
+def test_weighted_quantile_nan_cases(data, weights, quantile):
+    """Test that weighted_quantile returns NaN for degenerate cases."""
+    data_series = pd.Series(data)
+    weights_series = pd.Series(weights)
+    result = pv.weighted_quantile(data_series, weights_series, quantile)
+    assert np.isnan(result)
+
+
+@pytest.mark.parametrize(
+    "data,weights,quantile,expected",
+    [
+        # Data with inf values - function filters out inf but processes remaining data
+        ([1, np.inf, 3], [1, 1, 1], 0.5, 2.0),
+        ([1, -np.inf, 3], [1, 1, 1], 0.5, 2.0),
+        # NaN weights - function filters out rows with NaN weights
+        ([1, 2, 3], [1, np.nan, 1], 0.5, 2.0),
+        # Mix of NaN and inf - function filters out problematic rows
+        ([1, np.nan, np.inf, 4], [1, 1, 1, 1], 0.5, 2.5),
+    ],
+)
+def test_weighted_quantile_with_invalid_data(data, weights, quantile, expected):
+    """Test that weighted_quantile handles invalid data by filtering it out."""
+    data_series = pd.Series(data)
+    weights_series = pd.Series(weights)
+    result = pv.weighted_quantile(data_series, weights_series, quantile)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "column,threshold,n_outliers_allowed,expected_pass",
+    [
+        # Test cases that should PASS
+        ("stable_metric", 0.1, 0, True),  # Very stable values, low threshold
+        ("gradual_growth", 0.2, 0, True),  # Gradual 10% growth per year, 20% threshold
+        ("sudden_jump", 5.0, 0, True),  # Large jump but threshold accommodates it
+        ("volatile_metric", 0.3, 1, True),  # Volatile but allowing 1 outlier
```

Comment:
Is this exercising the outlier if volatile_metric never differs by more than 20%?